### PR TITLE
Unversion build artifacts

### PR DESCRIPTION
**Pull request by jasonkarns**
Carrying over conversation from https://github.com/twbs/bootstrap-rubygem/issues/78#issuecomment-291224077

Proposal: the sass and js files that are copied from https://github.com/twbs/bootstrap (hereafter "source repo") constitute build artifacts. My reasoning for this determination is that:

1. their existence is the result of an automated step that generates (or updates) them
1. they are _never_ intended to be edited or modified in this repo; their existence is solely for bundling in the gem itself
1. their presence in this repo is _not_ for version control (their changes are versioned in source repo)

It is well known and established that build artifacts should _not_ be committed to source control. [1] I will take it as assumed that this (fairly standard, IMO) opinion is shared, and so I will not enumerate its case, unless this core principle itself is debated.

Primary reason for removing source from this repo: reducing confusion for users. When using this gem, one naturally opens bug reports and issues wherever the source is. With _copies_ of the source in this repo, it is naturally assumed that these issues should be opened here. However, this is not the case. Presently, ~14% of issues on this repo were incorrectly opened here when they ought to have been opened on the source repo. (#9, #39, #40, #48, #49, #72, #77, #78) With the copies of the source removed, one must naturally find the actual source, which would lead them to the source repo itself.

Secondarily: the _actual_ source of this gem isn't the sass/js. As we've demonstrated, those are _generated_ files. The actual source of this repo is the updater task(s) that are somewhat hidden under tasks. Seeing the assets directory rather obscures the fact that the true source of this repo is the rake task itself. Rarely is the primary source of a repo a rake task or build script. So `tasks/` are naturally skipped over as users dive into a repo. The removal of the assets directory brings `lib/` and `tasks/` to the forefront, where they belong.

[1] there are, of course, cases where build artifacts should be versioned, but they are few and far between and have very specific rationale; typically catering to weird build processes that require the assets to be bundled.

Now that the Why is covered, here's the How:

1. removed the assets directory
2. gitignored the assets directory
3. update gemspec to define the gem's file listing without relying on git
4. allow rake update task to default the source sha as the value from lib/bootstrap/version.rb
5. update travis script to pull source before running tests

Nothing else changes. Steps for publishing continue to be:
1. run the updater rake task
2. run tests
3. gem build
4. gem publish

As an added bonus, this PR also adds the conventional bundler rake tasks for building, installing, and releasing ruby gems. Also rake's out-of-the-box `clean` and `clobber` tasks for removing intermediate/temp and build artifacts, respectively.

### CODE REVIEW

**Comment by jasonkarns** [comment]
Another option that hasn't been discussed here but should be on the table is adding the source repo as a subtree to this repo (or even as a submodule). That would be the most honest, since the sass/js would still be visible via this repo, but the relationship between the source git repo and this one is explicit (it would also simplify the rake update task and eliminate the need for going through the github API).

I don't have a super strong opinion on submodule vs subtree, but will say that submodules are generally preferred when development will likely occur _within the submodule_ with the intention of pushing changes back upstream. This is **not** the case here, so subtree would be the more appropriate solution, IMO. (The source repo is pulled in basically as a snapshot, not as a nested git repo.)

File: Rakefile [hunk 1]
```diff
@@ -61,6 +65,7 @@ end
 desc 'Update bootstrap from upstream'
 task :update, :branch do |t, args|
   require './tasks/updater'
+  args.with_defaults(:branch => Bootstrap::BOOTSTRAP_SHA)
```
Comment (glebm) [inline Rakefile]: The default should be the master branch.

**Comment by glebm** [comment]
> Another option that hasn't been discussed here but should be on the table is adding the source repo as a subtree to this repo (or even as a submodule).

The structure of this gem is different from the upstream repo, and the gem also generates files like `bootstrap-sprockets` and modifies a few other things, so I don't think using a subtree directly is an option.

**Comment by glebm** [comment]
A major issue I see with this implementation is that it does not allow using the gem via `gem 'bootstrap', git: '...'`.

**Pull request closed without merging**